#!/usr/bin/env python

PACKAGE = "singaboat_vrx"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("thrust_factor", double_t, 0, "Balancing factor for forward & reverse thrust forces", 2.5, 0, 5.0)
gen.add("drag_factor", double_t, 0, "Balancing factor for longitudinal & lateral drag forces", 1.5, 0, 5.0)
gen.add("v_x_gain", double_t, 0, "Linear velocity gain along X-axis", 0.45, 0, 1.0)
gen.add("v_y_gain", double_t, 0, "Linear velocity gain along Y-axis", 0.35, 0, 1.0)
gen.add("w_z_gain", double_t, 0, "Angular velocity gain about Z-axis", 1.2, 0, 5.0)
gen.add("half_beam", double_t, 0, "Half beam width of WAM-V", 1.027135, 0, 5.0)

gen.add("debug", bool_t, 0, "Flag to enable/disable debug messages", True)

exit(gen.generate(PACKAGE, "singaboat_vrx", "InverseKinematics"))
